test/powershell/Modules/Microsoft.Powershell.Host/Start-Transcript.Tests.ps1

```diff
@@ -132,19 +132,143 @@ Describe "Start-Transcript, Stop-Transcript tests" -tags "CI" {
-        $transcriptFilePath | Should FileContentMatch "PowerShell transcript end"
+        $transcriptFilePath | Should -FileContentMatch "Before Dispose"
+        $transcriptFilePath | Should -FileContentMatch "PowerShell transcript end"
     }
 
     It "Transcription should record native command output" {
         $script = {
             Start-Transcript -Path $transcriptFilePath
             hostname
-            Stop-Transcript }
+            Stop-Transcript
+        }
+
         & $script
-        Test-Path $transcriptFilePath | Should -BeTrue
 
+        Test-Path $transcriptFilePath | Should -BeTrue
         $machineName = [System.Environment]::MachineName
-        $transcriptFilePath | Should FileContentMatch $machineName
+        $transcriptFilePath | Should -FileContentMatch $machineName
+    }
+
+    It "Transcription should record Write-Information output when InformationAction is set to Continue" {
+        [String]$message = New-Guid
+        $script = {
+            Start-Transcript -Path $transcriptFilePath
+            Write-Information -Message $message -InformationAction Continue
+            Stop-Transcript
+        }
+
+        & $script
+
+        Test-Path $transcriptFilePath | Should -BeTrue
+        $transcriptFilePath | Should -Not -FileContentMatch "INFO: "
+        $transcriptFilePath | Should -FileContentMatch $message
+    }
+
+    It "Transcription should not record Write-Information output when InformationAction is set to SilentlyContinue" {
+        [String]$message = New-Guid
+        $script = {
+            Start-Transcript -Path $transcriptFilePath
+            Write-Information -Message $message -InformationAction SilentlyContinue
+            Stop-Transcript
+        }
+
+        & $script
+
+        Test-Path $transcriptFilePath | Should -BeTrue
+        $transcriptFilePath | Should -Not -FileContentMatch "INFO: "
+        $transcriptFilePath | Should -Not -FileContentMatch $message
+    }
+
+    It "Transcription should not record Write-Information output when InformationAction is set to Ignore" {
+        [String]$message = New-Guid
+        $script = {
+            Start-Transcript -Path $transcriptFilePath
+            Write-Information -Message $message -InformationAction Ignore
+            Stop-Transcript
+        }
+
+        & $script
+
+        Test-Path $transcriptFilePath | Should -BeTrue
+        $transcriptFilePath | Should -Not -FileContentMatch "INFO: "
+        $transcriptFilePath | Should -Not -FileContentMatch $message
+    }
+
+    It "Transcription should record Write-Information output in correct order when InformationAction is set to Inquire" {
+        [String]$message = New-Guid
+        $newLine = [System.Environment]::NewLine
+        $expectedContent = "$message$($newLine)Confirm$($newLine)Continue with this operation?"
+        $script = {
+            [System.Management.Automation.Internal.InternalTestHooks]::SetTestHook('ForcePromptForChoiceDefaultOption', $True)
```

Comment:
Seems like there should be an `AfterEach{}` that resets the test hook

Reply:
Done.